src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -109,24 +109,47 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 sb.Append("\\\"");
                 sb.Append(':');
 
-                var valuestr = payload[i] as string;
-
-                if( valuestr != null)
+                Type payloadType = payload[i].GetType();
+                if(payloadType == typeof(string))
+                {
+                    sb.Append("\\\"");
+                    minimalJsonserializer((string)payload[i], sb);
+                    sb.Append("\\\"");
+                }
+                else if(payloadType == typeof(byte[]))
                 {
                     sb.Append("\\\"");
-                    minimalJsonserializer(valuestr,sb);
+                    AppendByteArrayAsHexString(sb, (byte[])payload[i]);
                     sb.Append("\\\"");
                 }
                 else
                 {
                     sb.Append(payload[i].ToString());
                 }
-
             }
             sb.Append('}');
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        private static void AppendByteArrayAsHexString(StringBuilder builder, byte[] byteArray)
+        {
+            if(builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
```

Comment:
When will builder ever be null here?  Seems like this could just Debug.Assert rather than check/throw.